lib/identifiers/isbn.rb

```diff
@@ -13,17 +13,23 @@ def self.extract_isbn_as(str)
     end
 
     def self.extract_thirteen_digit_isbns(str)
-      str.gsub(/(?<=\d)[\p{Pd}\p{Zs}](?=\d)/, '').scan(REGEX_13).select { |isbn| valid_isbn_13?(isbn) }
+      str
+        .scan(REGEX_13)
+        .map { |isbn| isbn.gsub(/(?<=\d)[\p{Pd}\p{Zs}](?=\d)/, '') }
```

Comment:
We could probably get rid of the look-aheads and look-behinds here as hyphens and spaces can only appear between digits due to the previous `scan`.